### PR TITLE
Update LaTeXML recipe with missing modules

### PR DESCRIPTION
**Pull request by matteosecli**
Install all the necessary Perl modules for LaTeXML to work properly. Related issues: https://github.com/brucemiller/LaTeXML/issues/929 and https://github.com/Homebrew/homebrew-core/issues/22903.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Additional Info:
- I've tested the formula as indicated in point 5 of ["Create your pull request from a new branch"](https://github.com/Homebrew/brew/blob/master/docs/How-To-Open-a-Homebrew-Pull-Request.md#create-your-pull-request-from-a-new-branch) on a vanilla High Sierra install + Xcode 9 + Homebrew. The examples I've tried work as expected.
- I'd advise to use the HEAD version of LaTeXML, since version 0.8.2 may have some issues (https://github.com/brucemiller/LaTeXML/issues/934).

-----

### CODE REVIEW

**Comment by woodruffw** [comment]
Failures on Sierra and El Capital; I'm guessing this is a problem with the system-provided version of `libxml2`? If so, we probably need a conditional `depends_on` for MacOS <= Sierra.

```
running xml2-config...The installed version of libxml2 VERSION 2.9.4
 is not compatible with XML::LibXML (and probably buggy)!

You may continue at your own risk using 'perl Makefile.PL FORCE=1', but:

  - don't expect XML::LibXML to build or work correctly!
  - don't report errors!
  - don't send patches!

Check the README file for more information on versions
that are tested with XML::LibXML

failed
```

File: Formula/latexml.rb [hunk 1]
```diff
@@ -14,24 +14,277 @@ class Latexml < Formula
     sha256 "884426eb041a9fa05ba6ebc64c64f4ce76f7c10cab3c5c1b98bcce201831c9d2" => :mavericks
   end
 
+  requires_perl = []
+  requires_perl << "with-perl"
+  depends_on "perl"                             # Actually, Imagemagick already installs Perl
+  depends_on "imagemagick" => requires_perl     # Install ImageMagick with PerlMagick
```
Comment (ilovezfs) [inline Formula/latexml.rb]: This should trigger an audit failure which you've managed to defeat with the variable.

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
Going to pass on this as it adds significant complication and dependencies to a formula that is lightly used. But thanks for the PR nonetheless.

File: Formula/latexml.rb (hunk 1 above)
Comment (matteosecli) [reply Formula/latexml.rb]: You're right, but unfortunately ImageMagick needs to be installed with its own Perl module for LaTeXML to work. I am new to Homebrew recipes; do you have any suggestion for this problem?

**Comment by matteosecli** [comment]
> Failures on Sierra and El Capital; I'm guessing this is a problem with the system-provided version of libxml2? If so, we probably need a conditional depends_on for MacOS <= Sierra.

@woodruffw I think so; so far I just have a High Sierra box; I'm trying to get a Sierra one to test this option.

> Going to pass on this as it adds significant complication and dependencies to a formula that is lightly used.

@ilovezfs I agree that this complicate things, but as you see in the referenced issues:

- We all ❤️ Homebrew, and even if `latexml.rb` is lightly used the idea is to incentivize people to install LaTeXML via Homebrew itself.
- The level of complexity doesn't seem too far from https://github.com/Homebrew/homebrew-core/blob/master/Formula/rex.rb, which yet is in this repo; `rex.rb` pulls 43 Perl modules, the proposed `latexml.rb` pulls 44 of them.
- Unfortunately the current formula for LaTeXML doesn't work well and it lacks lots of dependencies. The only solution, so far, is to ask people to install all the missing Perl modules via cpan; however, it seems to me that this solution doesn't fit Homebrew's philosophy.

I really care about LaTeXML and I hope people will have a fully working formula out-of-the box, some day. Do you have any advice for this formula, so that I can work on it and get it accepted?

**Comment by ilovezfs** [comment]
@matteosecli you could set up a micro CPAN like znapzend did https://github.com/oetiker/znapzend/commit/73af9cdb8c43be3529d7fb9581ba3f854f5c5f16

**Comment by matteosecli** [comment]
Thanks for the tip, I'll definitely git it a try!

**Pull request closed without merging**